### ISSUE TITLE
Bug: Fixup stuck in Applying state

### ISSUE THREAD

**Issue by deepak2431** [sourcegraph/cody#1552]
When doing an edit fixup, the fixup is stuck in the applying state after applying the edits. "Cannot define property error is also shown along with it". 

See this for the exact behaviour:

https://github.com/sourcegraph/cody/assets/44617923/648930cc-f505-464e-903e-63f18e3ada5d

CC: @umpox @abeatrix

**Comment by deepak2431** [comment]
I tried to reproduce in the dev mode too, same issue. Looking over the fix of this.

**Comment by abeatrix** [comment]
@deepak2431 thanks for letting us know! @umpox has a wip that might have fixed it, can you give his branch a try and let us know if that helps? https://github.com/sourcegraph/cody/pull/1411

**Comment by deepak2431** [comment]
@abeatrix Thanks for letting me know about that. Let me give that branch a try.

**Comment by deepak2431** [comment]
@abeatrix It's the Same issue with that branch, too. This applies to the inline-chat edit too.


https://github.com/sourcegraph/cody/assets/44617923/36da9437-8d0d-4e07-8074-f8b5ca33ad4f



The auto-apply edits look good, but it is missing a case as when there's any error in the output, it just replaces the whole code with the error in the editor. 

https://github.com/sourcegraph/cody/assets/44617923/26e7b926-14e7-4837-9b15-2505f35ea1d7

**Comment by umpox** [comment]
Thanks @deepak2431, I'll investigate fixing this in the mentioned PR

**Comment by umpox** [comment]
@deepak2431 This should be fixed on the latest `main` now. Please take a look if you have chance!

The error was due to an incorrect assumption about a VS Code command (we get undefined instead of an empty array)

The second error looks unusual as it seems Cody is providing that as a valid response. I have added a PR to fix some cases where it'll catch an invalid response: https://github.com/sourcegraph/cody/pull/1561. I'll look further - may need to look at tweaking the prompt, I haven't been able to replicate this yet

**Comment by deepak2431** [comment]
@umpox Great!  Got it. I will test it again and let you know. 

I think the second error is when Cody responds, but it's something not code but like "I do not have enough context about the code" or a similar message.  So, we don't flag this as an error for now, and this is not a CORRECT response either. So, we need to find a way to flag this response in case of a fixup. LMK, What do you think?

I looked at your PR #1561, and that would display any error in case of an invalid response. I think so.

**Comment by deepak2431** [comment]
@umpox I tested it again on the latest same; the issue is still there for me. Can you please check it again once on your side?

**Comment by umpox** [comment]
Thanks @deepak2431

I think there is a possibility that this is dependant on the formatter on your machine, it could be timing out or taking a while. This is tricky for us to handle as we can't really control much here.

We can do better at showing this though, I've added a PR to show the formatting state and also add an option to skip:
https://github.com/sourcegraph/cody/pull/1561

I'd be curious to see if you now get stuck on "Formatting" which might help us narrow it down a bit (I still can't replicate this on `main` 🤔). We can look at adding the option to disable the formatting feature too

**Comment by deepak2431** [comment]
@umpox You are right. This was happening for me due to the formatter issue. I spent some time debugging it at the root level, so it's the `Python Black Formatter` that I am using that was causing this.

Also, after your recent change in the main, it shows stuck at Formatting instead of Applying. So, I'm not sure if this happens with all of the formatter, but yeah, I found the issue with the Python formatter, but with the TS formatter, it was not.

![iScreen Shoter - Code - 231102210409](https://github.com/sourcegraph/cody/assets/44617923/f0434bb3-8f43-42e8-8433-b18e91ab75a2)
